components/movementsensor/utils.go

```diff
@@ -90,6 +90,12 @@ type LastPosition struct {
 	mu           sync.Mutex
 }
 
+// LastCompassHeading store the last valid compass heading seen by the movement sensor.
```

Comment:
After reading through the code, I was going to write "why not just use something from `sync/atomic`, rather than reimplementing your own struct?" but looking through [the docs](https://pkg.go.dev/sync/atomic), and then [the feature requests](https://github.com/golang/go/issues/21996), it turns out Go doesn't provide built-in support for atomic float64's! Consider adding a comment to that effect ("This is really just an atomic float64, analogous to the atomic ints provided in the "sync/atomic" package.")